### ISSUE TITLE
pylint incorrect identification of "useless suppression"

### ISSUE THREAD

**Issue by Why-not-now** [PyCQA/pylint#7907]
### Bug description

```py
# pylint: disable=protected-access

_t = 1
a = _t
```

### Command used

```shell
pylint a.py --max-line-length=80 --enable=all "--disable=C0114, C0103, W0511"
```

### Pylint output

```shell
************* Module a
a.py:1:0: I0011: Locally disabling protected-access (W0212) (locally-disabled)
ha.py:1:0: I0021: Useless suppression of 'protected-access' (useless-suppression)
```

### Expected behavior

```
************* Module a
a.py:1:0: I0011: Locally disabling protected-access (W0212) (locally-disabled)
```
without the useless suppression warning

### Pylint version

```shell
pylint 2.15.8
astroid 2.12.13
Python 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)]
```

**Comment by Pierre-Sassoulas** [comment]
You can remove the ``# pylint: disable=protected-access`` line 1. A protected variable can be accessed in its scope without triggering a ``protected-access`` message. For a module level variable it means it can be accessed in its module, possibly its whole package.
